src/item.cpp

```diff
@@ -1260,6 +1262,115 @@ static void insert_separation_line( std::vector<iteminfo> &info )
     }
 }
 
+/*
+ * 0 based lookup table of accuracy - monster defense converted into number of hits per 1000
+ * attacks
+ * data painstakingly looked up at http://onlinestatbook.com/2/calculators/normal_dist.html
+ */
+static double hits_by_accuracy[41] = {
+     0.0,    0.1,   0.2,   0.3,   0.7, // -20 to -16
+     1.3,    2.6,   4.7,   8.2,  13.9, // -15 to -11
+    22.8,   35.9,  54.8,  80.8, 115.1, // -10 to -6
+    158.7, 211.9, 274.3, 344.6, 420.7, // -5 to -1
+    500,  // 0
+    579.3, 655.4, 725.7, 788.1, 841.3, // 1 to 5
+    884.9, 919.2, 945.2, 964.1, 977.2, // 6 to 10
+    986.1, 991.8, 995.3, 997.4, 998.7, // 11 to 15
+    999.3, 999.7, 999.8, 999.9, 1000.0 // 16 to 20
+};
+
+double item::effective_dps( const player &guy, monster &mon ) const
+{
+    const float mon_dodge = mon.get_dodge();
+    float base_hit = guy.get_hit_base() + guy.get_hit_weapon( *this );
+    base_hit *= std::max( 0.25f, 1.0f - guy.encumb( bp_torso ) / 100.0f );
+    float mon_defense = mon_dodge + mon.size_melee_penalty() / 5.0;
+    constexpr double hit_trials = 1000.0;
+    // a hit occurs when a normal distribution with a mean value of base_hit * 5 and a
+    // standard deviation of 25, minus 5 * monster defense, is greater than 0
+    // to shorten the lookup table, divide everything by 5 and simplify
+    // hit occurs when normal( acc - monster_defense, 5 ) > 0
+    const int rng_mean = std::max( std::min( static_cast<int>( base_hit - mon_defense ), 20 ),
+                                   -20 ) + 20;
+    double num_hits = hits_by_accuracy[ rng_mean ];
+    double moves_per_attack = guy.attack_speed( *this );
+    // attacks that miss do no damage but take time
+    double total_moves = ( hit_trials - num_hits ) * moves_per_attack;
+    double total_damage = 0.0;
+    double num_crits = hit_trials * guy.crit_chance( base_hit * 5.0, mon_dodge * 5.0, *this );
+    // critical hits are counted separately
+    num_hits -= num_crits;
+
+    // sum average damage past armor and return the number of moves required to achieve
+    // that damage
+    const auto calc_effective_damage = [ &, moves_per_attack]( int num_hits, bool crit,
+    const player & guy, monster & mon ) {
+        damage_instance base_damage;
+        guy.roll_all_damage( crit, base_damage, true, *this );
```

Comment:
It would be nice to have `roll_all_damage` un-randomized too, but that's probably trickier.

Reply:
It is, because that's what the second boolean returns - random values if false, average values if true.